pgjdbc/src/main/java/org/postgresql/largeobject/BlobInputStream.java

```diff
@@ -90,8 +92,30 @@ public BlobInputStream(LargeObject lo, int bsize, long limit) {
     // The very first read multiplies the last buffer size by two, so we divide by two to get
     // the first read to be exactly the initial buffer size
     this.lastBufferSize = INITIAL_BUFFER_SIZE / 2;
+
+    // Trying to seek past the max size throws an exception, which skip() should avoid
+    long maxBlobSize;
+    try {
+      // initialise current position for mark/reset
+      this.absolutePosition = lo.tell64();
+      // https://github.com/postgres/postgres/blob/REL9_3_0/src/include/storage/large_object.h#L76
+      maxBlobSize = (long) Integer.MAX_VALUE * LO_BLOCK_SIZE;
```

Comment:
Because that is a breaking change to the behaviour of `skip(long)`.
https://github.com/pgjdbc/pgjdbc/pull/3148#discussion_r1513081689

What it currently does is skip to the end of the current input for all values larger than the remaining input. This includes returning 0 if it's already at the end.

I had another idea recently. We could provide a setter to allow users to change it if they have built postgres with a custom block size.

Reply:
I suggest we defer the call to the database. I do not see value in artificially adding a limit on the driver's side.

Reply:
So I therefore need to split the PR so that the bug fix doesn't have to wait for v43?

Reply:
Turns out older postgres behaves differently, looking at the 8.4 test failure. I will investigate further.